node-notes-cli/app.js

```diff
@@ -0,0 +1,52 @@
+import * as fs from 'node:fs/promises';
+
+try {
+  if (process.argv[2] === 'read') {
+    read();
+  }
+  if (process.argv[2] === 'create') {
+    create();
+  }
+  if (process.argv[2] === 'update') {
+    update();
+  }
+  if (process.argv[2] === 'delete') {
+    remove();
+  }
+} catch (err) {
+  console.error('Error:', err.message);
+}
+
+async function read() {
+  const content = await fs.readFile('data.json');
```

Comment:
You repeat this read/parse code many times in here. It would be nice to remove the duplication.

Similarly, you repeat the stringify/write code many times. That duplication should also be removed.